### DISCUSSION TITLE
Version 7.0

### DISCUSSION THREAD

**Discussion by Akascape** [Akascape/CTkDesigner-Support#26] [New Update]
# What's new in version 7.0?

- Added new grid background for better widget placement
![image](https://github.com/user-attachments/assets/4f1b29a1-ef4b-41ea-86fe-bffb4d2cdf6c)

> Theme>Toggle Grid
> Shortcut: Ctrl+Q 
> Adjust the cells through the + and - button 

- New Undo feature (experimental)
> Currently recovers deleted widgets and placements
> Shortcut: Ctrl+z

- Fixed duplicate issue when moved to top 
- Now we can fix the parameter window to right side when opened, helpful if you window size is large. Just click the new gear icon in the bottom of param window
- Now we have ability to link a slider and label, where the label text changes to the slider value
Applicable widgets: CTkSlider, CTkRangeSlider, CTkMeter, CTkColorPicker
![image](https://github.com/user-attachments/assets/dfa4c2f6-8d65-48f8-a6b8-c82fa42e0c0e)

- Added Ctrl+C and Ctrl+V for widgets
- Added ability to duplicate whole frame and it's subwidgets
- Added feathering (corner softness) in CTkDraw widget (ctkdlib)
- Now the lock parameter can be saved in Json
- Fixed CTkListbox parameter window issue when open through the menu
- Now the slider value and textbox text can be exported
- Added scrollable frame orientation
- And many more minor improvements and bug fixes...

Make sure you update ctkdlib to 7.0

**Comment by nordicjg** [comment]
Thank you very much for this application! It is very helpful.
After working a bit with it I would like to share my findings:

**Bugs**

- “Fix above all” cannot be reverted/changed
- After “Fix in frame” widgets can’t be moved properly (the widget jumps out of view. In some cases the widget can’t be edited/deleted anymore)
- After “Edit Theme” the “Switch Theme” has no effect

**Enhancements/Wishes**

- Icon support (e.g. with material design font, not images)
- Multiselection of widgets
  - Align left/right/top/bottom
  - Move by drag and drop
- Treeview of all widgets in page.
  - Widgets “fix in frame” are second level
  - Change of ordering
  - Multiselection
- Vertical segmented button
- Handling of dialogs
- possibility to define custom styles for widgets (instead of repeatedly manually change styles fg_color, bg_color …). In general the code is cluttered with a lot of styling statements.
  - Button (e.g. Cancel  bg=red, icon)
  - Label (e.g. Heading1  font-size 16 / bold)
- Export Pages to separate files/classes

**Reply by Akascape** [reply]
@nordicjg Great ideas, will work on these soon. I will try to implement the possible features and give the next update soon...
